airflow/www/app.py

```diff
@@ -81,10 +82,18 @@ def create_app(config=None, testing=False):
 
     flask_app.config["PERMANENT_SESSION_LIFETIME"] = timedelta(minutes=settings.get_session_lifetime_config())
     flask_app.config.from_pyfile(settings.WEBSERVER_CONFIG, silent=True)
-    flask_app.config["APP_NAME"] = conf.get(section="webserver", key="instance_name", fallback="Airflow")
     flask_app.config["TESTING"] = testing
     flask_app.config["SQLALCHEMY_DATABASE_URI"] = conf.get("database", "SQL_ALCHEMY_CONN")
 
+    instance_name = conf.get(section="webserver", key="instance_name", fallback="Airflow")
+    instance_name_has_markup = conf.getboolean(
+        section="webserver", key="instance_name_has_markup", fallback=False
+    )
+    if instance_name_has_markup:
+        instance_name = re.sub(r"<[^<]+?>", "", instance_name)
```

Comment:
Do we want to also un-escape the value? (for `&amp;` etc) Maybe it’s better to use an HTML parser for this. Or maybe that’s an overkill. No idea.

Reply:
> Do we want to also un-escape the value? (for &amp; etc) 

Something like https://docs.python.org/3/library/html.html#html.unescape ?

From the docs:
> Convert all named and numeric character references (e.g. `&gt;`, `&#62;`, `&#x3e;`) in the string s to the corresponding Unicode characters. This function uses the rules defined by the HTML 5 standard for both valid and invalid character references, and the [list of HTML 5 named character references](https://docs.python.org/3/library/html.entities.html#html.entities.html5).

Reply:
Parsing html with regex can be tricky. Please see this answer https://stackoverflow.com/a/4869782/2610955 and also https://blog.codinghorror.com/parsing-html-the-cthulhu-way/. The question has several options and also discusses handling `&amp;` . Since beautifulsoup is already a dependency maybe it can be used.

```python
python      
Python 3.10.6 (main, Aug  2 2022, 15:11:28) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import bs4
>>> from bs4 import BeautifulSoup
>>> b = BeautifulSoup("<b>Bold Site Title Test</b>")          
>>> b.get_text()
'Bold Site Title Test'
```

Reply:
Sorry, I had a devel installation and beautifulsoup is a devel dependency.

Reply:
Yeah that's why I wonder if this would be an overkill. It's not terribly difficult to implement text extraction with http.parser (stdlib) but whether it's worthwhile is still susceptive.

Reply:
Used @uranusjr as suggestion for bc21a02. I really didn't know where to put the method. Is there a better place?